pages/docs/faq.mdx

```diff
@@ -41,6 +41,32 @@ In practice, it's important to note that you should typically only run Effects a
 
 For a more detailed comparison, you can refer to the [Effect vs fp-ts](fp-ts) documentation.
 
+### A Closer Look at `flatMap`
+
+Many JavaScript / TypeScript engineers are familiar with the term `flatMap` due to it's [presence as a method on the `Array` prototype](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap). Therefore it may be confusing to see `flatMap` methods exported from many of the modules that Effect provides.
+
+The `flatMap` operation can actually be used to describe a more generic data transformation. Let's consider for a moment a generic data type that we will call `F`. `F` will also be a container for elements, so we can further refine our representation of `F` to `F<A>` which states that `F` holds some information about some data of type `A`.
+
+If we have an `F<A>` and we want to transform to an `F<B>`, we could use the `map` operation:
+
+```
+map: F<A> -> (A -> B) -> F<B>
```

Comment:
This looks too much Haskell-y IMO, I would use a more TypeScript-y signature here, like
```
map: <A, B>(fa: F<A>, f: (a: A) => B) => F<B>
```

Reply:
Done 👍